monkey/infection_monkey/exploit/hadoop.py

```diff
@@ -78,12 +82,21 @@ def exploit(self, url, command):
             [random.choice(string.ascii_lowercase) for _ in range(self.RAN_STR_LEN)]  # noqa: DUO102
         )
         payload = self._build_payload(app_id, rand_name, command)
+
+        if self._is_interrupted():
+            self._set_interrupted()
+            return False
+
         resp = requests.post(
             posixpath.join(url, "ws/v1/cluster/apps/"), json=payload, timeout=LONG_REQUEST_TIMEOUT
         )
         return resp.status_code == 202
 
     def check_if_exploitable(self, url):
+        if self._is_interrupted():
+            self._set_interrupted()
+            return False
+
```

Comment:
`check_if_exploitable()` gets called from within a loop in `add_vulnerable_urls()`. That loop uses an interruptable iterator, so I think this check is redundant.